src/LightSpeed/userFunction.js

```diff
@@ -53,33 +53,100 @@ const COMMON_SCOPE = {
     },
 };
 
-const COMMON_SPAWN_SCOPE = {
-    // Keep this in sync with SpawnScope class in
-    // hack-toy-apps/com.endlessm.LightSpeed/userScope.js
-    ticksSinceSpawn: 0,
+// Set method that checks type for number properties of objects
+function propNumFactory(obj, propName, valName) {
```

Comment:
I'm finding it difficult to understand what `valName` is for... it seems to me the value should be stored somewhere on the object, or does this already do that somehow?

Reply:
I'm still having trouble puzzling out how `valName` works ... maybe a comment would help?

Optional, I think you could also use this to define the `x` and `y` properties on `Point` to make it less verbose.